### PR TITLE
Add torus creation method.

### PR DESCRIPTION
**Pull request by MaxJPRey**
It is now possible to create a torus with the modeler.

Fix issue #658 .

### CODE REVIEW

**Review by maxcapodi78 — APPROVED** [review]
LGTM

File: pyaedt/modeler/Primitives3D.py [hunk 1]
```diff
@@ -345,6 +346,70 @@ def create_sphere(self, position, radius, name=None, matname=None):
...
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.Object3d.Object3d`
+            3D object.
+
+        References
+        ----------
+
+        >>> oEditor.CreateTorus
```
Comment (akaszynski) [inline pyaedt/modeler/Primitives3D.py:380]: Is this necessary?

File: pyaedt/modeler/Primitives3D.py [hunk 2]
```diff
@@ -345,6 +346,70 @@ def create_sphere(self, position, radius, name=None, matname=None):
...
+
+        Examples
+        --------
+
```
Comment (akaszynski) [inline pyaedt/modeler/Primitives3D.py:384]:
```suggestion
    Create a torus named ``"mytorus"`` about the Z axis with a major
    radius of 1, minor radius of 0.5, and a material of ``"copper"``.

```

File: pyaedt/modeler/Primitives3D.py [hunk 3]
```diff
@@ -345,6 +346,70 @@ def create_sphere(self, position, radius, name=None, matname=None):
...
+        >>> from pyaedt import Hfss
+        >>> hfss = Hfss()
+        >>> origin = [0,0,0]
+        >>> torus = hfss.modeler.primitives.create_torus(origin, major_radius=1,
+        ...                                             minor_radius=0.5, axis="Z",
+        ...                                             name="mytorus", material_name="copper")
```
Comment (akaszynski) [inline pyaedt/modeler/Primitives3D.py:390]:
```suggestion
        >>> origin = [0, 0, 0]
        >>> torus = hfss.modeler.primitives.create_torus(origin, major_radius=1,
        ...                                              minor_radius=0.5, axis="Z",
        ...                                              name="mytorus", material_name="copper")
```

File: pyaedt/modeler/Primitives3D.py [hunk 4]
```diff
@@ -345,6 +346,70 @@ def create_sphere(self, position, radius, name=None, matname=None):
...
+
+        """
+        assert len(center) == 3, "Center argument must be a valid 3 elements List."
+        assert major_radius > 0 and minor_radius > 0, "Both major and minor radius must be greater than 0."
+        assert minor_radius < major_radius, "Major radius must be greater than minor radius."
```
Comment (akaszynski) [inline pyaedt/modeler/Primitives3D.py:395]: Do not use assertions, rather raise errors. See:
https://medium.com/@jadhavmanoj/python-what-is-raise-and-assert-statement-c3908697bc62

> Python’s assert statement is a debugging aid, not a mechanism for handling run-time errors. The goal of using assertions is to let developers find the likely root cause of a bug more quickly. An assertion error should never be raised unless there’s a bug in your program.

File: pyaedt/modeler/Primitives3D.py [hunk 5]
```diff
@@ -345,6 +346,70 @@ def create_sphere(self, position, radius, name=None, matname=None):
...
+
+        vArg1 = ["NAME:TorusParameters"]
+        vArg1.append("XCenter:="), vArg1.append(x_center)
+        vArg1.append("YCenter:="), vArg1.append(y_center)
+        vArg1.append("ZCenter:="), vArg1.append(z_center)
+        vArg1.append("MajorRadius:="), vArg1.append(major_radius)
+        vArg1.append("MinorRadius:="), vArg1.append(minor_radius)
+        vArg1.append("WhichAxis:="), vArg1.append(axis)
+        vArg2 = self._default_object_attributes(name=name, matname=material_name)
+        new_object_name = _retry_ntimes(10, self._oeditor.CreateTorus, vArg1, vArg2)
```
Comment (akaszynski) [inline pyaedt/modeler/Primitives3D.py:410]: Parameters should not be capitalized.

```suggestion
        varg1 = ["NAME:TorusParameters"]
        varg1.append("XCenter:="), varg1.append(x_center)
        varg1.append("YCenter:="), varg1.append(y_center)
        varg1.append("ZCenter:="), varg1.append(z_center)
        varg1.append("MajorRadius:="), varg1.append(major_radius)
        varg1.append("MinorRadius:="), varg1.append(minor_radius)
        varg1.append("WhichAxis:="), varg1.append(axis)
        varg2 = self._default_object_attributes(name=name, matname=material_name)
        new_object_name = _retry_ntimes(10, self._oeditor.CreateTorus, varg1, varg2)
```

File: _unittest/test_08_Primitives3D.py [hunk 6]
```diff
@@ -901,3 +910,11 @@ def test_66_assign_material(self):
     def test_67_cover_lines(self):
         P1 = self.aedtapp.modeler.primitives.create_polyline([[0, 1, 2], [0, 2, 3], [2, 1, 4]], close_surface=True)
         assert self.aedtapp.modeler.cover_lines(P1)
+
+    @pyaedt_unittest_check_desktop_error
+    def test_68_create_torus(self):
+        torus = self.create_copper_torus()
```
Comment (akaszynski) [inline _unittest/test_08_Primitives3D.py:916]: Test for `pytest.raises` after accepting below suggestion.

File: pyaedt/modeler/Primitives3D.py (hunk 1 above)
Comment (MaxJPRey) [reply pyaedt/modeler/Primitives3D.py:380]: Yes it is.

File: pyaedt/modeler/Primitives3D.py (hunk 4 above)
Comment (MaxJPRey) [reply pyaedt/modeler/Primitives3D.py]: Done

**Pull request merged**